pkg/vsphere/guestinfo/encode_test.go

```diff
@@ -0,0 +1,409 @@
+// Copyright 2016 VMware, Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package guestinfo
+
+import (
+	"net/url"
+	"os/exec"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/vmware/govmomi/vim25/types"
+)
+
+// [BEGIN] SLIMMED VERSION of github.com/vmware/vic/metadata
+type Common struct {
+	ExecutionEnvironment string
+
+	ID string `vic:"0.1" scope:"hidden" key:"id"`
+
+	Name string `vic:"0.1" scope:"hidden" key:"name"`
+
+	Notes string `vic:"0.1" scope:"hidden" key:"notes"`
+}
+
+type ContainerVM struct {
+	Common `vic:"0.1" scope:"read-only" key:"common"`
+
+	Version string `vic:"0.1" scope:"hidden" key:"version"`
+
+	Aliases map[string]string
+
+	Interaction url.URL
+
+	AgentKey []byte
+}
+
+type ExecutorConfig struct {
+	Common `vic:"0.1" scope:"read-only" key:"common"`
+
+	Sessions map[string]SessionConfig `vic:"0.1" scope:"hidden" key:"sessions"`
+
+	Key []byte
+}
+
+type Cmd struct {
+	Path string `vic:"0.1" scope:"hidden" key:"path"`
+
+	Args []string `vic:"0.1" scope:"hidden" key:"args"`
+
+	Env []string `vic:"0.1" scope:"hidden" key:"env"`
+
+	Dir string `vic:"0.1" scope:"hidden" key:"dir"`
+
+	Cmd *exec.Cmd `vic:"0.1" scope:"hidden" key:"cmd"`
+}
+
+type SessionConfig struct {
+	Common `vic:"0.1" scope:"hidden" key:"common"`
+
+	Cmd Cmd `vic:"0.1" scope:"hidden" key:"cmd"`
+
+	Tty bool `vic:"0.1" scope:"hidden" key:"tty"`
+}
+
```

Comment:
recommend that we have a test that has some json tags for example - a case where there are tags that are not 'vic'. Don't bother about a mix of both vic and something else.